packages/vue/src/components/SpAssets/SpAssets.vue

```diff
@@ -57,26 +177,155 @@ export default defineComponent({
       )
     })
 
+    let isShowMore = computed(() => {
+      if (!balances.value.length) {
+        return
+      }
+
+      if (!state.searchQuery) {
+        return filteredBalanceList.value.length < balances.value.concat(state.demoDenoms).length
+      } else {
+        return filteredBalanceList.value.length < filteredArrayLength
+      }
+    })
+
+    const filteredBalanceList = computed(() => {
+      let searchArray = balances.value.length ? balances.value.concat(state.demoDenoms) : [],
+          searchString = state.searchQuery
+
+      if (!searchString) {
+        return searchArray.slice(0, state.displayLimit);
+      }
+
+      searchString = searchString.trim().toLowerCase()
+
+      searchArray = searchArray.filter((item) => {
+        if (item.denom.toLowerCase().includes(searchString)) {
+          return item
+        }
+      })
+      filteredArrayLength = searchArray.length
+
+      // Return an array with the filtered data.
+      return searchArray.slice(0, state.displayLimit)
+    })
+
+    const showMore = () => {
+      state.displayLimit += state.displayLimit
+    }
+
+    const resetSearch = () => {
+      state.searchQuery = "";
+      searchInput.value?.focus();
+    }
+
+    watch(
+      () => state.searchQuery,
+      (searchQuery, prevSearchQuery) => {
+        if (!searchQuery) {
+          state.displayLimit = props.resultLimit
+        }
+      }
+    )
+
     // lh
     onMounted(async () => {
       await queryAllBalances({
-        params: { address: props.address },
+        params: { address: props.address},
+        query: {"pagination.reverse": false},
         options: { all: true, subscribe: true }
+      }).then(() => {
+        setTimeout(() => uiState.isAssetsLoading = false, 1500);
```

Comment:
to remove

Reply:
The timeout is removed, but the callback must be left to change the loading state.